django/bossspatialdb/test/int_test_to_black_view.py

```diff
@@ -0,0 +1,298 @@
+from rest_framework.test import force_authenticate
+from rest_framework import status
+import numpy as np
+
+from bossspatialdb.views import CutoutToBlack, Cutout
+
+from bosscore.test.setup_db import DjangoSetupLayer
+
+import redis
+import blosc
+
+
+version = settings.BOSS_VERSION
+
+
+class CutoutToBlackView8BitTests(APITestCase):
+    layer = DjangoSetupLayer
+
+    def test_uint8_cutout_to_black(self):
+        """Cutout_to_black integration test with uint8 data"""
+        test_mat = np.random.randint(1, 254, (16, 128, 128))
+        test_mat = test_mat.astype(np.uint8)
+        h = test_mat.tobytes()
+        bb = blosc.compress(h, typesize=8)
+
+        # Create request
+        factory = APIRequestFactory()
+        request = factory.post('/' + version + '/cutout/col1/exp1/channel1/0/0:128/0:128/0:16/', bb,
+                               content_type='application/blosc')
+        # log in user
+        force_authenticate(request, user=self.user)
+
+        # Make request
+        response = Cutout.as_view()(request, collection='col1', experiment='exp1', channel='channel1',
+                                    resolution='0', x_range='0:128', y_range='0:128', z_range='0:16', t_range=None)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        factory = APIRequestFactory()
+        request = factory.put('/' + version + '/cutout/to_black/col1/exp1/channel1/0/0:64/0:64/0:16/',
+                               content_type='application/blosc-python')
+        # log in user
+        force_authenticate(request, user=self.user)
+
+        # Delete a portion of Data
+        response = CutoutToBlack.as_view()(request, collection='col1', experiment='exp1', channel='channel1',
+                                    resolution='0', x_range='0:64', y_range='0:64', z_range='0:16', t_range=None)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Create Request to get original posted data
+        request = factory.get('/' + version + '/cutout/col1/exp1/channel1/0/0:128/0:128/0:16/',
+                              accepts='application/blosc')
+
+        # log in user
+        force_authenticate(request, user=self.user)
+
+        # Make request
+        response = Cutout.as_view()(request, collection='col1', experiment='exp1', channel='channel1',
+                                    resolution='0', x_range='0:128', y_range='0:128', z_range='0:16', t_range=None).render()
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+        # Decompress
+        raw_data = blosc.decompress(response.content)
+        data_mat = np.fromstring(raw_data, dtype=np.uint8)
+        data_mat = np.reshape(data_mat, (16, 128, 128), order='C')
+
+        # Test for data equality (what you put in is what you got back!)
+        test_mat[:,0:64, 0:64] = np.zeros((16,64,64), dtype=np.uint8)
+        np.testing.assert_array_equal(data_mat, test_mat)
```

Comment:
For all of these, it might be worthwhile to test the other half of the cube to verify that it wasn't "blacked" out.